core/src/main/java/com/github/kaktushose/jda/commands/message/MessageResolver.java

```diff
@@ -0,0 +1,53 @@
+package com.github.kaktushose.jda.commands.message;
+
+import com.github.kaktushose.jda.commands.message.i18n.I18n;
+import com.github.kaktushose.jda.commands.message.emoji.EmojiResolver;
+import org.jspecify.annotations.Nullable;
+
+import java.util.Locale;
+import java.util.Map;
+
+/// The MessageResolver combines [I18n#localize(Locale, String, I18n.Entry...)] and
+/// [EmojiResolver#resolve(String)].
+///
+/// It will first do localization with help of [I18n] and then apply the returned string
+/// to [EmojiResolver] to resolve Unicode aliases and application emojis.
+///
+/// Please note that this class is a helper and doesn't have own resolving logic, it's more of a pipeline
+/// to [EmojiResolver] and [I18n]. It is not intended to be directly used by end users but part of the public api
+/// to allow manual execution of the frameworks resolving logic for dynamic values if needed.
+public class MessageResolver {
+    private final I18n i18n;
+    private final EmojiResolver emojiResolver;
+
+    public MessageResolver(I18n i18n, EmojiResolver emojiResolver) {
+        this.i18n = i18n;
+        this.emojiResolver = emojiResolver;
+    }
+
+    /// Please visit the docs of [I18n#localize(Locale, String, I18n.Entry...)]
+    /// and [EmojiResolver#resolve(String)]
```

Comment:
lazy docs :D

Reply:
Well, this class shouldn't be often used by endusers + I see no point in practically copying docs together